pandora-client-web/src/components/wardrobe/modules/wardrobeModuleLockSlot.tsx

```diff
@@ -468,3 +512,58 @@ function PasswordInput({
 		</>
 	);
 }
+
+function TimerInput({
+	onChange,
+	timer,
+	pendingAttempt = false,
+}: {
+	onChange: (newValue: number) => void;
+	timer: Immutable<NonNullable<LockSetup['timer']>>;
+	pendingAttempt?: boolean;
+	showInvalidWarning?: boolean;
+}) {
+	const id = useId();
+
+	const [hours, setHours] = useState<number>(0);
+	const [minutes, setMinutes] = useState<number>(0);
+
+	const maxHours = useMemo(() => {
+		return timer.maxMinutes / 60;
+	}, [timer]);
+
+	const maxMinutes = useMemo(() => {
+		return timer.maxMinutes < 59 ? timer.maxMinutes : 59;
+	}, [timer]);
```

Comment:
This approach technically works, but can behave in slightly unexpected ways at times. The fewer `useState`s you have, the better. In this case I would recommend passing the current `timer` value from `WardrobeLockSlotUnlocked` as a property and calculating `hours` and `minutes` from it.
Then `setHours` and `setMinutes` should be implemented in a way that they result in calling `onChange` with the new timer as a single value. (this then causes the state value to update in the parent component, propagating here, and causing the new calculation of `hours` and `minutes` from it)